content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/components/authoring/editor/pagepreview/renderconditions/isproductpage/isproductpage.jsp

```diff
@@ -0,0 +1,57 @@
+<%--
+
+    Copyright 2019 Adobe. All rights reserved.
+    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License. You may obtain a copy
+    of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software distributed under
+    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+    OF ANY KIND, either express or implied. See the License for the specific language
+    governing permissions and limitations under the License.
+
+--%><%
+%><%@include file="/libs/granite/ui/global.jsp" %><%
+%><%@page session="false"
+        import="com.adobe.granite.ui.components.rendercondition.RenderCondition,
+                  com.adobe.granite.ui.components.rendercondition.SimpleRenderCondition,
+				  org.apache.commons.lang.StringUtils,
+				  org.apache.sling.api.resource.NonExistingResource,
+				  org.apache.sling.api.resource.Resource,
+				  org.apache.sling.api.resource.ResourceResolver,
+                  com.adobe.granite.ui.components.Config,
+                  com.day.cq.wcm.api.Page,
+                  com.day.cq.wcm.api.PageManager" %><%
+%><%
+    final Config cfg = cmp.getConfig();
+    final String path = cmp.getExpressionHelper().getString(cfg.get("path", String.class));
+    boolean decision = isProductDetailPage(path, resourceResolver);
+    request.setAttribute(RenderCondition.class.getName(), new SimpleRenderCondition(decision));
+%><%!
+    private boolean isProductDetailPage(String path, ResourceResolver resourceResolver) {
+        if(StringUtils.isBlank(path)) {
+            return false;
+        }
+
+        Resource resource = resourceResolver.resolve(path);
+        if (resource instanceof NonExistingResource) {
+            return false;
+        }
+
+        Page page = resourceResolver.adaptTo(PageManager.class).getPage(resource.getPath()) ;
+        if (page == null) {
+            return false;
+        }
+
+        Resource pageContent = page.getContentResource();
+        if ( pageContent == null) {
+            return false;
+        }
+
+        String cqTemplate = pageContent.getValueMap().get("cq:template", String.class);
+        if (StringUtils.isBlank(cqTemplate)) {
+            return false;
+        }
+        return cqTemplate.endsWith("/settings/wcm/templates/product-page");
```

Comment:
Is there no better way to detect the product detail page? This creates a hard dependency on the project structure created by our archetype. If somebody does not use the archetype or change / rename this afterward it might break.
Maybe checking if either the page jcr:content or the template jcr:content contains the product detail page is a better approach.

Reply:
@mhaack , I was also thinking what would be a good way to do that. The relative location settings/wcm/templates is quite common for the sites that I can see in Quickstart so I though this would work.  I admit that improvement would be good for this and several other cases where we simply rely on convention.

Reply:
I'm not worried about the relative location `settings/wcm/templates` which is given be template mechanism. But the name `product-page`, which can be anything a customer changes to and this is what the hard link to Venia structure is. What about customers not starting with Venia content structure and using their own templates. This approach will fail.

While checking for CIF product component (sling super resource type) on the page will be pretty save check because than indicates they use CIF core components and in particular our product component, hence the component which is a key part for this feature.

Reply:
@mhaack, now I see what you mean. Make sense. I've improved the code.

Reply:
+1 like that approach much more.